hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -77,6 +80,11 @@ public static Object getNestedFieldVal(GenericRecord record, String fieldName, b
 
       // return, if last part of name
       if (i == parts.length - 1) {
+
+        if (isLogicalTypeDate(valueNode, part)) {
```

Comment:
Should we do this conversion out in the caller of this method instead?

Reply:
This method seems to me like the right central place to perform this check and conversion. If we check the accessors of this method, it is used by Hudi to retrieve the key values for its metadata fields. It might be more standard to have Hudi treat Date as the actual Date string instead of a Long, across all its keys for its internal usage. It would create a lot of confusion otherwise and accessors of this functions will have to take care of adding this check.

Also, we would again have to re-write the same logic/loop again on client side to check for logical type. Because this function basically returns only the value.

Reply:
Hmmm.. just feel that now, this method is doing multiple things and those get tricky over time.. Simple compromise.. please pull this  conversion code to its own helper method, that we can invoke from here.. that way, over time, more conversions are just added to that helper., while this methods just focusses on getting a nested value .